app/src/main/java/de/westnordost/streetcomplete/overlays/surface/PathSurfaceOverlayForm.kt

```diff
@@ -0,0 +1,448 @@
+import de.westnordost.streetcomplete.osm.surface.COMMON_SPECIFIC_UNPAVED_SURFACES
+import de.westnordost.streetcomplete.osm.surface.CyclewayFootwaySurfaces
+import de.westnordost.streetcomplete.osm.surface.CyclewayFootwaySurfacesWithNote
+import de.westnordost.streetcomplete.osm.surface.GENERIC_AREA_SURFACES
+import de.westnordost.streetcomplete.osm.surface.GROUND_SURFACES
+import de.westnordost.streetcomplete.osm.surface.SingleSurface
+import de.westnordost.streetcomplete.osm.surface.SingleSurfaceWithNote
+import de.westnordost.streetcomplete.osm.surface.Surface
+import de.westnordost.streetcomplete.osm.surface.SurfaceAndNote
+import de.westnordost.streetcomplete.osm.surface.SurfaceInfo
+import de.westnordost.streetcomplete.osm.surface.SurfaceMissing
+import de.westnordost.streetcomplete.osm.surface.SurfaceMissingWithNote
+import de.westnordost.streetcomplete.osm.surface.applyTo
+import de.westnordost.streetcomplete.osm.surface.asItem
+import de.westnordost.streetcomplete.osm.surface.commonSurfaceObject
+import de.westnordost.streetcomplete.osm.surface.createSurfaceStatus
+import de.westnordost.streetcomplete.osm.surface.shouldBeDescribed
+import de.westnordost.streetcomplete.osm.surface.toItems
+import de.westnordost.streetcomplete.overlays.AbstractOverlayForm
+import de.westnordost.streetcomplete.overlays.AnswerItem
+import de.westnordost.streetcomplete.overlays.IAnswerItem
+import de.westnordost.streetcomplete.quests.surface.DescribeGenericSurfaceDialog
+import de.westnordost.streetcomplete.util.ktx.nonBlankTextOrNull
+import de.westnordost.streetcomplete.view.image_select.DisplayItem
+import de.westnordost.streetcomplete.view.image_select.ImageListPickerDialog
+import de.westnordost.streetcomplete.view.image_select.ItemViewHolder
+
+class PathSurfaceOverlayForm : AbstractOverlayForm() {
+    override val contentLayoutResId = R.layout.fragment_overlay_path_surface_select
+    private val binding by contentViewBinding(FragmentOverlayPathSurfaceSelectBinding::bind)
+
+    private val itemsPerRow = 2
+    /** items to display. May not be accessed before onCreate */
+    val items: List<DisplayItem<Surface>> = (COMMON_SPECIFIC_PAVED_SURFACES + COMMON_SPECIFIC_UNPAVED_SURFACES + GROUND_SURFACES + GENERIC_AREA_SURFACES).toItems()
+    private val cellLayoutId: Int = R.layout.cell_icon_select_with_label_below
+    private var originalSurfaceStatus: SurfaceInfo? = null
+    private var isSegregatedLayout = false
+
+    private var selectedStatusForMainSurface: DisplayItem<Surface>? = null
+        set(value) {
+            field = value
+            updateSelectedCell()
+        }
+    private var selectedStatusForCyclewaySurface: DisplayItem<Surface>? = null
+        set(value) {
+            field = value
+            updateSelectedCell()
+        }
+    private var selectedStatusForFootwaySurface: DisplayItem<Surface>? = null
+        set(value) {
+            field = value
+            updateSelectedCell()
+        }
```

Comment:
I feel there is **a lot** of duplication in UI code for this class due to all the three surface select buttons and the associated text field to work exactly the same.

I already de-duplicated the layout code for those a bit, but it may be worth to look into whether putting functionality of this select button into an `ImageSelectButtonViewController`.

Reply:
I agree, but I had no good idea how to do it (maybe for reasons that are not applicable anymore - but maybe there are still some distinctions between them)

Reply:
Then let's defer that for now until I get to reviewing the forms properly.

Reply:
I see that interface code was improved by @westnordost - thanks!